packages/remix-dev/config.ts

```diff
@@ -529,12 +529,6 @@ export async function resolveConfig(
     entryServerFile = `entry.server.${serverRuntime}.tsx`;
   }
 
-  if (isSpaMode && appConfig.future?.unstable_lazyRouteDiscovery === true) {
-    throw new Error(
-      "You can not use `future.unstable_lazyRouteDiscovery` in SPA Mode (`ssr: false`)"
-    );
-  }
```

Comment:
I think this was unintentionally left in here from a first pass - we decided up front that SPA mode would just not have fog of war so its never enabled at run time, so no need to fail the build (which would be a problem in RR v7 anyway)

```js
export function isFogOfWarEnabled(future: FutureConfig, isSpaMode: boolean) {
  return future.v3_lazyRouteDiscovery === true && !isSpaMode;
}
```